src/BlockchainReader/ClaimController.ts

```diff
@@ -15,86 +17,77 @@ export class ClaimController {
+    this.bitcoinCore = bitcoinCore
     this.configuration = configuration
     this.collection = this.db.collection('blockchainReader')
   }
 
   async scanBlock(blockHeight: number): Promise<void> {
     const logger = this.logger.child({ method: 'scanBlock' })
 
-    logger.trace({ blockHeight }, 'Retrieving Block Hash...')
+    logger.debug({ blockHeight }, 'Retrieving Block Hash')
 
-    const blockHash = await this.insightHelper.getBlockHash(blockHeight)
+    const blockHash = await this.bitcoinCore.getBlockHash(blockHeight)
 
     logger.trace(
       {
         blockHeight,
         blockHash,
       },
-      'Block Hash retrieved successfully. Retrieving Raw Block...'
+      'Block Hash retrieved successfully. Retrieving Raw Block'
     )
 
-    const block = await this.insightHelper.getBlock(blockHash)
-    const poetTimestamps: ReadonlyArray<PoetTimestamp> = block.transactions
-      .map(getPoetTimestamp)
-      .filter(_ => _)
-      .map(_ => ({
-        ..._,
-        blockHeight,
-        blockHash,
-      }))
-    const transactionIds = block.transactions.map(_ => _.id)
+    const block: Block = await this.bitcoinCore.getBlock(blockHash, GetBlockVerbosity.Transactions)
 
-    const matchingPoetTimestamps = poetTimestamps
-      .filter(this.poetTimestampNetworkMatches)
-      .filter(this.poetTimestampVersionMatches)
+    const anchors = blockToPoetAnchors(block)
+
+    const anchorPrefixAndVersionMatchConfiguration = anchorPrefixAndVersionMatch(
+      this.configuration.poetNetwork,
+      this.configuration.poetVersion
+    )
 
-    const unmatchingPoetTimestamps = poetTimestamps.filter(_ => !matchingPoetTimestamps.includes(_))
+    const matchingAnchors = filter(anchorPrefixAndVersionMatchConfiguration, anchors)
+    const unmatchingAnchors = reject(anchorPrefixAndVersionMatchConfiguration, anchors)
 
     logger.trace(
       {
         blockHeight,
         blockHash,
-        matchingPoetTimestamps,
-        unmatchingPoetTimestamps,
+        matchingAnchors,
+        unmatchingAnchors,
       },
-      'Raw Block retrieved and scanned successfully'
+      'Block retrieved and scanned successfully'
     )
 
     await this.collection.updateOne(
       { blockHeight },
       {
         $set: {
           blockHash,
-          transactionIds,
-          matchingPoetTimestamps,
-          unmatchingPoetTimestamps,
+          matchingAnchors,
+          unmatchingAnchors,
         },
       },
       { upsert: true }
     )
 
-    if (matchingPoetTimestamps.length) await this.messaging.publishPoetTimestampsDownloaded(matchingPoetTimestamps)
+    if (matchingAnchors.length) await this.messaging.publishPoetTimestampsDownloaded(matchingAnchors)
   }
 
-  /**
-   * @returns {Promise<number>} The highest block height from the database, or null if no block has been processed.
-   */
-  async findHighestBlockHeight(): Promise<number> {
+  async findHighestBlockHeight(): Promise<number | null> {
```

Comment:
I don't sure is good practice return a `null`

Reply:
Agreed. I personally don't see a problem with returning undefined or null as long as the signature makes it explicit, but in the future I'd like to try using an Option monad instead.